Error creating annotation from empty annotator, overlaid on element, and with `.groupby`

Error when creating a new annotation from an empty annotator overlaid on an element and when .groupby has been set

<details> <summary> Code </summary>

```python
# ensure there is no .db in the cwd

from holonote.annotate import Annotator
from holonote.app import PanelWidgets
import panel as pn; pn.extension()
import holoviews as hv; hv.extension('bokeh')
from holonote.annotate.connector import SQLiteDB

annotator = Annotator({"height": float, "width": float}, fields=["type"],
                      connector=SQLiteDB(filename=':memory:'))
annotator.groupby = "type"
annotator_widgets = PanelWidgets(annotator)

pn.Column(annotator_widgets, annotator * hv.Image([], ['width', 'height'])).servable()
```

Then try to create and commit a new annotation 

</details>

https://github.com/holoviz/holonote/assets/6613202/125c7210-0c14-405a-955e-df504bb34ab9


<details> <summary> Traceback </summary>

```bash
WARNING:param.OverlayPlot04752: :NdOverlay   [type] is empty and will be skipped during plotting
Traceback (most recent call last):
File "/Users/droumis/opt/miniconda3/envs/neuro-deep-image-stack/lib/python3.12/site-packages/holoviews/plotting/util.py", line 293, in get_plot_frame
return map_obj[key]
~~~~~~~^^^^^
File "/Users/droumis/opt/miniconda3/envs/neuro-deep-image-stack/lib/python3.12/site-packages/holoviews/core/spaces.py", line 1216, in __getitem__
val = self._execute_callback(*tuple_key)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/Users/droumis/opt/miniconda3/envs/neuro-deep-image-stack/lib/python3.12/site-packages/holoviews/core/spaces.py", line 983, in _execute_callback
retval = self.callback(*args, **kwargs)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/Users/droumis/opt/miniconda3/envs/neuro-deep-image-stack/lib/python3.12/site-packages/holoviews/core/spaces.py", line 552, in __call__
return self.callable()
^^^^^^^^^^^^^^^
File "/Users/droumis/opt/miniconda3/envs/neuro-deep-image-stack/lib/python3.12/site-packages/holoviews/util/__init__.py", line 1038, in dynamic_operation
key, obj = resolve(key, kwargs)
^^^^^^^^^^^^^^^^^^^^
File "/Users/droumis/opt/miniconda3/envs/neuro-deep-image-stack/lib/python3.12/site-packages/holoviews/util/__init__.py", line 1027, in resolve
return key, map_obj[key]
~~~~~~~^^^^^
File "/Users/droumis/opt/miniconda3/envs/neuro-deep-image-stack/lib/python3.12/site-packages/holoviews/core/spaces.py", line 1219, in __getitem__
self._cache(tuple_key, val)
File "/Users/droumis/opt/miniconda3/envs/neuro-deep-image-stack/lib/python3.12/site-packages/holoviews/core/spaces.py", line 1292, in _cache
self[key] = val
~~~~^^^^^
File "/Users/droumis/opt/miniconda3/envs/neuro-deep-image-stack/lib/python3.12/site-packages/holoviews/core/ndmapping.py", line 566, in __setitem__
self._add_item(key, value, update=False)
File "/Users/droumis/opt/miniconda3/envs/neuro-deep-image-stack/lib/python3.12/site-packages/holoviews/core/ndmapping.py", line 165, in _add_item
self._item_check(dim_vals, data)
File "/Users/droumis/opt/miniconda3/envs/neuro-deep-image-stack/lib/python3.12/site-packages/holoviews/core/ndmapping.py", line 984, in _item_check
raise AssertionError(f"{self.__class__.__name__} must only contain one type of object, not both {type(data).__name__} and {self.type.__name__}.")
AssertionError: DynamicMap must only contain one type of object, not both Overlay and NdOverlay.
```

</details>

I'm not sure how to move forward. I can't seem to catch the error condition using either caplog or capsys. Here is the test I was trying. @hoxbro any tips?

```python
def test_groupby_with_overlay_from_empty_annotator(annotator_range2d, caplog, capsys):
    annotator = annotator_range2d
    annotator.groupby = "description"
    bounds = (-1, -1, 1, 1)
    data = np.array([[0, 1], [1, 0]])
    img = hv.Image(data, kdims=["x", "y"], bounds=bounds)

    plot = annotator * img
    hv.render(plot)

    annotator.set_regions(x=(-0.15, 0.15), y=(-0.25, 0.25))

    with caplog.at_level(logging.WARNING):
        try:
            annotator.add_annotation(description="Test")
        except AssertionError as e:
            assert "DynamicMap must only contain one type of object" not in str(e), f"Found unexpected AssertionError: {e}"

    captured = capsys.readouterr()
    print("Captured stdout:", captured.out)
    print("Captured stderr:", captured.err)

    # Log messages check
    log_messages = [record.message for record in caplog.records]
    print("Captured log messages:", log_messages)

    assert not any("DynamicMap must only contain one type of object" in message for message in log_messages), \
        "Found an unexpected AssertionError log message"
```

And here is the output of pytest.. wich passes despite showing the captured stderr showing the error:

<details> <summary> pytest output </summary>

```python
pytest -s holonote/tests/test_annotators_element.py
========================================================= test session starts =========================================================
platform darwin -- Python 3.10.14, pytest-8.1.1, pluggy-1.4.0
rootdir: /Users/droumis/src/holonote
configfile: pyproject.toml
plugins: nbval-0.11.0
collected 16 items

holonote/tests/test_annotators_element.py ..........WARNING:param.OverlayPlot12867: :NdOverlay   [description] is empty and will be skipped during plotting
WARNING:param.OverlayPlot12867: :NdOverlay   [description] is empty and will be skipped during plotting
WARNING:param.OverlayPlot12867: :NdOverlay   [description] is empty and will be skipped during plotting
Captured stdout: Traceback (most recent call last):
  File "/Users/droumis/opt/miniconda3/envs/holonote-dev/lib/python3.10/site-packages/holoviews/plotting/util.py", line 293, in get_plot_frame
    return map_obj[key]
  File "/Users/droumis/opt/miniconda3/envs/holonote-dev/lib/python3.10/site-packages/holoviews/core/spaces.py", line 1217, in __getitem__
    val = self._execute_callback(*tuple_key)
  File "/Users/droumis/opt/miniconda3/envs/holonote-dev/lib/python3.10/site-packages/holoviews/core/spaces.py", line 984, in _execute_callback
    retval = self.callback(*args, **kwargs)
  File "/Users/droumis/opt/miniconda3/envs/holonote-dev/lib/python3.10/site-packages/holoviews/core/spaces.py", line 552, in __call__
    return self.callable()
  File "/Users/droumis/opt/miniconda3/envs/holonote-dev/lib/python3.10/site-packages/holoviews/util/__init__.py", line 1037, in dynamic_operation
    key, obj = resolve(key, kwargs)
  File "/Users/droumis/opt/miniconda3/envs/holonote-dev/lib/python3.10/site-packages/holoviews/util/__init__.py", line 1026, in resolve
    return key, map_obj[key]
  File "/Users/droumis/opt/miniconda3/envs/holonote-dev/lib/python3.10/site-packages/holoviews/core/spaces.py", line 1220, in __getitem__
    self._cache(tuple_key, val)
  File "/Users/droumis/opt/miniconda3/envs/holonote-dev/lib/python3.10/site-packages/holoviews/core/spaces.py", line 1293, in _cache
    self[key] = val
  File "/Users/droumis/opt/miniconda3/envs/holonote-dev/lib/python3.10/site-packages/holoviews/core/ndmapping.py", line 566, in __setitem__
    self._add_item(key, value, update=False)
  File "/Users/droumis/opt/miniconda3/envs/holonote-dev/lib/python3.10/site-packages/holoviews/core/ndmapping.py", line 165, in _add_item
    self._item_check(dim_vals, data)
  File "/Users/droumis/opt/miniconda3/envs/holonote-dev/lib/python3.10/site-packages/holoviews/core/ndmapping.py", line 986, in _item_check
    raise AssertionError(f"{self.__class__.__name__} must only contain one type of object, not both {type(data).__name__} and {self.type.__name__}.")
AssertionError: DynamicMap must only contain one type of object, not both Overlay and NdOverlay.


Captured stderr:
Captured log messages: [':NdOverlay   [description] is empty and will be skipped during plotting', ':NdOverlay   [description] is empty and will be skipped during plotting', ':NdOverlay   [description] is empty and will be skipped during plotting']
......

========================================================= 16 passed in 5.29s ==========================================================
```

</details>

The error message is located in stdout, and is emitted from https://github.com/holoviz/holoviews/blob/febe07506bda2c433ac31938ce4c5aa62335559d/holoviews/plotting/util.py#L299. 

``` python
def test_groupby_with_overlay_from_empty_annotator(annotator_range2d, capsys):
    # Test for https://github.com/holoviz/holonote/issues/119
    annotator = annotator_range2d
    annotator.groupby = "description"
    bounds = (-1, -1, 1, 1)
    data = np.array([[0, 1], [1, 0]])
    img = hv.Image(data, kdims=["x", "y"], bounds=bounds)

    plot = annotator * img
    hv.render(plot)

    annotator.set_regions(x=(-0.15, 0.15), y=(-0.25, 0.25))
    annotator.add_annotation(description="Test")

    captured = capsys.readouterr()
    bad_output = "AssertionError: DynamicMap must only contain one type of object, not both Overlay and NdOverlay."
    assert bad_output not in captured.out
```

--- a/holonote/annotate/display.py
+++ b/holonote/annotate/display.py
@@ -519,6 +519,9 @@ class AnnotationDisplay(param.Parameterized):
         highlighters = {opt: self._selected_dim_expr(v[0], v[1]) for opt, v in highlight.items()}
         indicator = indicator.opts(*self.style.indicator(**highlighters))
 
+        if len(indicator.data) == 0:
+            return hv.NdOverlay({0: self._make_empty_element()})
+
         return indicator.overlay() if self.annotator.groupby else hv.NdOverlay({0: indicator})
 
     def _selected_dim_expr(self, selected_value, non_selected_value) -> hv.dim:

--- a/holonote/tests/test_annotators_element.py
+++ b/holonote/tests/test_annotators_element.py
@@ -251,6 +251,25 @@ def test_groupby_visible(cat_annotator):
         next(iter_indicator)
 
 
+def test_groupby_with_overlay_from_empty_annotator(annotator_range2d, capsys):
+    # Test for https://github.com/holoviz/holonote/issues/119
+    annotator = annotator_range2d
+    annotator.groupby = "description"
+    bounds = (-1, -1, 1, 1)
+    data = np.array([[0, 1], [1, 0]])
+    img = hv.Image(data, kdims=["x", "y"], bounds=bounds)
+
+    plot = annotator * img
+    hv.render(plot)
+
+    annotator.set_regions(x=(-0.15, 0.15), y=(-0.25, 0.25))
+    annotator.add_annotation(description="Test")
+
+    captured = capsys.readouterr()
+    bad_output = "AssertionError: DynamicMap must only contain one type of object, not both Overlay and NdOverlay."
+    assert bad_output not in captured.out
+
+
 def test_multiply_overlay(annotator_range1d):
     el1 = hv.Curve([], kdims=["TIME"])
     el2 = hv.Curve([], kdims=["TIME"])

